app/networkapi/opportunities/models.py

```diff
@@ -1,5 +1,42 @@
 from django.db import models
+from datetime import datetime, timezone
+from slugify import slugify
+from adminsortable.models import SortableMixin
+import os
 
 
-class Opportunity(models.Model):
-    pass
+def opportunity_image_path(instance, filename):
+    return 'images/opportunity/{name}_{timestamp}{ext}'.format(
+        name=slugify(instance.name, max_length=300),
+        timestamp=str(
+            int((
+                datetime.now(tz=timezone.utc) -
+                datetime(1970, 1, 1, tzinfo=timezone.utc)
+            ).total_seconds())
+        ),
+        ext=os.path.splitext(filename)[1]
+    )
+
+
+class Opportunity(SortableMixin):
+    name = models.CharField(max_length=300)
+    header = models.TextField(max_length=5000)
+    description = models.TextField(max_length=5000)
+    link_text = models.CharField(max_length=300)
```

Comment:
let's call this `link_label` instead since that's what the frontend refers to it as.